packages/salesforcedx-vscode-core/test/jest/commands/deployExecutor.test.ts

```diff
@@ -162,74 +163,139 @@ describe('Deploy Executor', () => {
-    expect(sfdxCommandletExecutorClearSpy).toHaveBeenCalled();
   });
 
-  it('should unlock queue on failure', async () => {
-    // Arrange
-    const mockUnlock = jest.fn();
-    const unlockSpy = jest
-      .spyOn(DeployQueue, 'get')
-      .mockReturnValue({ unlock: mockUnlock } as any);
-    const executor = new TestDeployExecutor(
-      'testDeploy',
-      'force_source_deploy_with_sourcepath_beta'
-    );
-    // Act
-    await (executor as any).postOperation();
+  describe('postOperation', () => {
+    let mockUnlock: any;
+    let unlockSpy: any;
+    let setPropertiesForFilesDeployMock: any;
+    let getInstanceSpy: any;
+    let getPackageDirectoryPathsSpy: any;
+    let createOutputSpy: any;
+    let appendLineSpy: any;
+    beforeEach(() => {
+      setPropertiesForFilesDeployMock = jest.fn();
+      getInstanceSpy = jest
+        .spyOn(PersistentStorageService, 'getInstance')
+        .mockReturnValue({
+          setPropertiesForFilesDeploy: setPropertiesForFilesDeployMock
+        } as any);
+      getPackageDirectoryPathsSpy = jest
+        .spyOn(SfdxPackageDirectories, 'getPackageDirectoryPaths')
+        .mockResolvedValue('path/to/foo' as any);
+      createOutputSpy = jest
+        .spyOn(TestDeployExecutor.prototype as any, 'createOutput')
+        .mockReturnValue('path/to/foo' as any);
+      appendLineSpy = jest
+        .spyOn(channelService, 'appendLine')
+        .mockImplementation(jest.fn());
+      mockUnlock = jest.fn();
+      unlockSpy = jest
+        .spyOn(DeployQueue, 'get')
+        .mockReturnValue({ unlock: mockUnlock } as any);
+      DeployRetrieveExecutor.errorCollection = MockErrorCollection as any;
+    });
+
+    it('should clear errors on success', async () => {
+      // Arrange
+      const mockDeployResult = {
+        response: {
+          status: 'Succeeded'
+        }
+      };
+      const deployRetrieveExecutorClearSpy = jest.spyOn(
+        DeployRetrieveExecutor.errorCollection,
+        'clear'
+      );
+      SfdxCommandletExecutor.errorCollection = MockErrorCollection as any;
+      const sfdxCommandletExecutorClearSpy = jest.spyOn(
+        SfdxCommandletExecutor.errorCollection,
+        'clear'
+      );
+
+      const executor = new TestDeployExecutor(
+        'testDeploy',
+        'force_source_deploy_with_sourcepath_beta'
+      );
+
+      // Act
+      await (executor as any).postOperation(mockDeployResult);
+
+      // Assert
+      expect(getInstanceSpy).toHaveBeenCalled();
+      expect(getPackageDirectoryPathsSpy).toHaveBeenCalled();
+      expect(createOutputSpy).toHaveBeenCalled();
+      expect(appendLineSpy).toHaveBeenCalled();
+      expect(setPropertiesForFilesDeployMock).toHaveBeenCalledWith(
+        mockDeployResult
+      );
+      expect(deployRetrieveExecutorClearSpy).toHaveBeenCalled();
+      expect(sfdxCommandletExecutorClearSpy).toHaveBeenCalled();
+      expect(unlockSpy).toHaveBeenCalled();
+      expect(mockUnlock).toHaveBeenCalled();
```

Comment:
mockUnlock is returnedValue of unlockSpy. Do we need this expect statement? Also mockUnlock should be a boolean, right?